### PR TITLE
Add two new data types to the bulk loader

### PR DESCRIPTION
**Pull request by JeffSchering**
This is for [PLT-6937: Bulk Importing of Direct/Group channels and posts. #6761](https://github.com/mattermost/platform/pull/6761)

### CODE REVIEW

File: source/deployment/bulk-loading-data-format.rst [hunk 1]
```diff
@@ -14,7 +14,11 @@ Channel
 User
   Optional. If present, User objects must occur after the Team and Channel objects in the file and before any Post objects. Each User object defines the teams and channels that the user is a member of. If the corresponding teams and channels are not in the data file, then they must exist in the Mattermost database.
 Post
-  Optional. If present, Post objects must occur after all other objects in the file. Each Post object defines the team, the channel, and the username of the user who posted the message. If the corresponding team, channel, or user are not in the data file, then they must exist in the Mattermost database.
+  Optional. If present, Post objects must occur after the last User object but before any DirectChannel objects. Each Post object defines the team, the channel, and the username of the user who posted the message. If the corresponding team, channel, or user are not in the data file, then they must exist in the Mattermost database.
```
Comment (lindy65) [inline source/deployment/bulk-loading-data-format.rst]: "... or user are..." > "... or user is..."

File: source/deployment/bulk-loading-data-format.rst [hunk 2]
```diff
@@ -689,3 +695,131 @@ Fields of the Post object
...
+      <th class="head">Field name</th>
+      <th class="head">Type</th>
+      <th class="head">Description</th>
+      <th class="head">Validated</th>
+      <th class="head">Mandatory</th>
+    </tr>
+    <tr class="row-odd">
+      <td valign="middle">members</td>
+      <td valign="middle">array</td>
+      <td>Must contain a list of members, with a minumum of two usernames and a maximum of eight usernames.</td>
```
Comment (lindy65) [inline source/deployment/bulk-loading-data-format.rst]: "minumum" > "minimum"

File: source/deployment/bulk-loading-data-format.rst [hunk 3]
```diff
@@ -689,3 +695,131 @@ Fields of the Post object
...
+      <th class="head">Field name</th>
+      <th class="head">Type</th>
+      <th class="head">Description</th>
+      <th class="head">Validated</th>
+      <th class="head">Mandatory</th>
+    </tr>
+    <tr class="row-odd">
+      <td valign="middle">channel_members</td>
+      <td valign="middle">array</td>
+      <td>Must contain a list of members, with a minumum of two usernames and a maximum of eight usernames.</td>
```
Comment (lindy65) [inline source/deployment/bulk-loading-data-format.rst]: "minumum" > "minimum"

File: source/deployment/bulk-loading-data-format.rst [hunk 4]
```diff
@@ -689,3 +695,131 @@ Fields of the Post object
...
+      <th class="head">Type</th>
+      <th class="head">Description</th>
+      <th class="head">Validated</th>
+      <th class="head">Mandatory</th>
+    </tr>
+    <tr class="row-odd">
+      <td valign="middle">members</td>
+      <td valign="middle">array</td>
+      <td>Must contain a list of members, with a minumum of two usernames and a maximum of eight usernames.</td>
+      <td align="center" valign="middle">Yes</td>
```
Comment (grundleborg) [inline source/deployment/bulk-loading-data-format.rst]: This should be a No* where the footnote is: Not validated, but an error will be thrown in apply mode if any of the users do not exist.

File: source/deployment/bulk-loading-data-format.rst (hunk 1 above)
Comment (JeffSchering) [inline source/deployment/bulk-loading-data-format.rst]: I think "are" is correct. It agrees in number with the list (team, channel, or user) and with the plural personal pronoun "they".

File: source/deployment/bulk-loading-data-format.rst (hunk 2 above)
Comment (JeffSchering) [inline source/deployment/bulk-loading-data-format.rst]: ok

File: source/deployment/bulk-loading-data-format.rst (hunk 3 above)
Comment (JeffSchering) [inline source/deployment/bulk-loading-data-format.rst]: ok

File: source/deployment/bulk-loading-data-format.rst (hunk 4 above)
Comment (JeffSchering) [inline source/deployment/bulk-loading-data-format.rst]: Done.

File: source/deployment/bulk-loading-data-format.rst (hunk 1 above)
Comment (lindy65) [inline source/deployment/bulk-loading-data-format.rst]: I'll go with your preference but, as far as I'm aware, if "or" is used in a list - each thing listed is considered single - not part of a group.  However, if "and" is used, then the list of items becomes a group and can be plural.

**Pull request merged**